### PR TITLE
Add `--exclude=<hooks>` argument to `wp cron event run`

### LINKED ISSUE #55: Feature Request: cron event --exclude
Related to #27 and its possible solution #38, but along a different line.

**Here's what I'm doing.** 

I have a large number of crons on my site because I use an RSS Importer to import posts from dozens of feeds. The RSS Importer checks the feeds once per (hour, 2 hours, etc., it varies). I do try to stagger the crons so they don't all run at once, but given all the other WP core related crons and crons for future publishing posts, and etc., etc., I usually have at least half a dozen crons that are ready to run `Now`.

I have WP Cron disabled and on my server I run the following:

```
*/5  *  *  *  *       cd /home/public_html && /usr/local/bin/wp cron event run 'publish_future_post' --due-now
*/5  *  *  *  *       cd /home/public_html && /usr/local/bin/wp cron event run 'fetch_single_feed_hook' --due-now
*/8  *  *  *  *       cd /home/public_html && /usr/local/bin/wp cron event run  --due-now
```

**The problem:**

Sometimes `wp cron event run 'fetch_single_feed_hook' --due-now` has not finished firing by the time the `wp cron event run  --due-now` starts, and then I get duplicate results. This is the problem described in #27.

**My suggestion:**

If there was an `--exclude` option to the cron event command, I could run something like:

`wp cron event run  --due-now --exclude=fetch_single_feed_hook,publish_future_post`

Then only the other crons due `Now` would run, and not the ones that potentially are already running separately.

### PR DESCRIPTION
**Pull request by rutviksavsani**
Fix #55  
 - This pull request adds the `--exclude` flag to the `wp cron event run` command.
 - The flag accepts a list of comma separated hooks that will be excluded from the list of events.
 
 Working: 
 - Without the exclude flag 
 
image

- With the exclude flag

image

### CODE REVIEW

File: src/Cron_Event_Command.php [hunk 1]
```diff
@@ -233,6 +236,20 @@ public function run( $args, $assoc_args ) {
...
+
+		if ( ! empty( $exclude ) ) {
+			$exclude = explode( ',', $exclude );
+		}
+
+		$events = array_filter(
+			$events,
+			function ( $event ) use ( $args, $exclude ) {
+				return ( empty( $args ) || in_array( $event->hook, $args, true ) ) &&
+					( empty( $exclude ) || ! in_array( $event->hook, $exclude, true ) );
```
Comment (danielbachhuber) [inline src/Cron_Event_Command.php:249]: This is pretty hard to read. Can you make it easier to read?

File: src/Cron_Event_Command.php [hunk 2]
```diff
@@ -210,6 +210,9 @@ public function schedule( $args, $assoc_args ) {
 	 * [--due-now]
 	 * : Run all hooks due right now.
 	 *
+	 * [--exclude=<hooks>]
+	 * : Exclude events of a comma-separated list of hooks.
```
Comment (danielbachhuber) [inline src/Cron_Event_Command.php:214]:
```suggestion
	 * : Comma-separated list of events to exclude.
```

**Comment by wojsmol** [comment]
@rutviksavsani Please update `Issue: #55` to `Fix #55` so Github will auto close related issue on PR merge.

File: src/Cron_Event_Command.php (hunk 1 above)
Comment (rutviksavsani) [reply src/Cron_Event_Command.php]: Yes, Made the logic simpler.

**Comment by rutviksavsani** [comment]
Hi @danielbachhuber 
I have addressed the feedback as suggested.
Thanks!

**Review by danielbachhuber — APPROVED** [review]

**Comment by danielbachhuber** [comment]
> I have addressed the feedback as suggested.

@rutviksavsani Awesome, thanks! I added one more test with f879760018a99996b89714be646a0ef154f44082

Nice work on this 😊

**Pull request merged**